File: nodes/array-filter/array-filter.process.py
async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  if not isinstance(array, list):
    array = [array]
    
  value = inputs.get("value")
  condition = settings.get("condition", "equals")
  case_sensitive = settings.get("case_sensitive", True)
  
  def compare(item):
    if condition == "equals":
      if isinstance(item, str) and isinstance(value, str) and not case_sensitive:
        return item.lower() == value.lower()
      return item == value
      
    elif condition == "not_equals":
      if isinstance(item, str) and isinstance(value, str) and not case_sensitive:
        return item.lower() != value.lower()
      return item != value
      
    elif condition == "greater_than":
      return item > value
      
    elif condition == "less_than":
      return item < value
      
    elif condition == "contains":
      if isinstance(item, str) and isinstance(value, str):
        return value.lower() in item.lower() if not case_sensitive else value in item
      return False
      
    elif condition == "not_contains":
      if isinstance(item, str) and isinstance(value, str):
        return value.lower() not in item.lower() if not case_sensitive else value not in item
      return True
      
    elif condition == "exists":
      return item is not None
      
    elif condition == "not_exists":
      return item is None
      
    return True
  
  return {
    "array": list(filter(compare, array))
  } 

File: nodes/array-filter/array-filter.process.js
export default async ({inputs, settings, config}) => {

  const array = Array.isArray(inputs.array) ? inputs.array : [inputs.array];
  const value = inputs.value;
  const condition = settings.condition || "equals";
  const caseSensitive = settings.case_sensitive !== false;

  const filtered = array.filter(item => {
    switch (condition) {
      case "equals":
        if (typeof item === "string" && typeof value === "string" && !caseSensitive) {
          return item.toLowerCase() === value.toLowerCase();
        }
        return item === value;
        
      case "not_equals":
        if (typeof item === "string" && typeof value === "string" && !caseSensitive) {
          return item.toLowerCase() !== value.toLowerCase();
        }
        return item !== value;
        
      case "greater_than":
        return item > value;
        
      case "less_than":
        return item < value;
        
      case "contains":
        if (typeof item === "string" && typeof value === "string") {
          return caseSensitive ? 
            item.includes(value) : 
            item.toLowerCase().includes(value.toLowerCase());
        }
        return false;
        
      case "not_contains":
        if (typeof item === "string" && typeof value === "string") {
          return caseSensitive ? 
            !item.includes(value) : 
            !item.toLowerCase().includes(value.toLowerCase());
        }
        return true;
        
      case "exists":
        return item !== null && item !== undefined;
        
      case "not_exists":
        return item === null || item === undefined;
        
      default:
        return true;
    }
  });

  return {
    array: filtered
  };
}; 